pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test: TestRepeatableReadLogic_select_for_share_write_buffering failed

pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test.TestRepeatableReadLogic_select_for_share_write_buffering [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21268380?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21268380?buildTab=artifacts#/) on master @ [f94af11c7119504a40a0b18ac5d6fc4d95e52571](https://github.com/cockroachdb/cockroach/commits/f94af11c7119504a40a0b18ac5d6fc4d95e52571):


```
=== RUN   TestRepeatableReadLogic_select_for_share_write_buffering
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/717ae83f7b098c7772eaf712614a8a34/logTestRepeatableReadLogic_select_for_share_write_buffering4131835359
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    test_server_shim.go:150: cluster virtualization disabled due to issue: #156124 (expected label: C-bug)
    logictestbase.go:589: statement/query skipped with reason: unsupported configuration local-repeatable-read, statement/query only supports local-read-committed (no issue given)
    logic.go:3237: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11162/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_share_write_buffering:71: SELECT database_name, schema_name, table_name, lock_key_pretty, lock_strength, durability, replace(isolation_level, ' ', '_') AS isolation_level, granted, contended FROM crdb_internal.cluster_locks
        expected:
            database_name  schema_name  table_name  lock_key_pretty   lock_strength  durability    isolation_level  granted  contended
            test           public       t           /Table/106/1/1/0  Shared         Unreplicated  REPEATABLE_READ  true     true
            test           public       t           /Table/106/1/1/0  Exclusive      Unreplicated  REPEATABLE_READ  false    true
            test           public       t           /Table/106/1/1/0  Shared         Unreplicated  REPEATABLE_READ  true     true
            
        but found (query options: "colnames,retry,rowsort" -> ignore the following ordering of rows) :
            database_name  schema_name  table_name  lock_key_pretty   lock_strength  durability  isolation_level  granted  contended
            test           public       t           /Table/106/1/1/0  Shared         Replicated  REPEATABLE_READ  true     true
            test           public       t           /Table/106/1/1/0  Exclusive      Replicated  REPEATABLE_READ  false    true
            test           public       t           /Table/106/1/1/0  Shared         Replicated  REPEATABLE_READ  true     true
            
[01:56:04] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11162/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_share_write_buffering: 14 statements
[01:56:04] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11162/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_share_write_buffering with config local-repeatable-read: 14 tests, 1 failures
    logic.go:4657: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11162/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_share_write_buffering:79: error while processing
    logic.go:4657: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11162/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/select_for_share_write_buffering:79: too many errors encountered, skipping the rest of the input
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/717ae83f7b098c7772eaf712614a8a34/logTestRepeatableReadLogic_select_for_share_write_buffering4131835359
--- FAIL: TestRepeatableReadLogic_select_for_share_write_buffering (45.97s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRepeatableReadLogic_select_for_share_write_buffering.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-61713

--- a/pkg/sql/logictest/testdata/logic_test/select_for_share_write_buffering
+++ b/pkg/sql/logictest/testdata/logic_test/select_for_share_write_buffering
@@ -13,6 +13,12 @@ SET CLUSTER SETTING kv.transaction.write_buffering.transformations.get.enabled =
 statement ok
 SET CLUSTER SETTING kv.transaction.write_buffering.transformations.scans.enabled = true;
 
+# Pin the buffer size to avoid a small metamorphic max_buffer_size causing an
+# immediate flush, which would disable write buffering and its lock durability
+# transformations for the rest of the transaction.
+statement ok
+SET CLUSTER SETTING kv.transaction.write_buffering.max_buffer_size = '2KiB';
+
 statement ok
 SET kv_transaction_buffered_writes_enabled = true;
 

